client/gutenboarding/store/reducer.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { Reducer } from 'redux';
+import { combineReducers } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import ActionTypes from './action-types';
+import { SetSiteTypeAction } from './actions';
+
+const siteType: Reducer< null | string, SetSiteTypeAction > = ( state = null, action ) => {
```

Comment:
In the long run, we probably want the reducer not to work on `string`, but on a discrete selection of site types, see https://github.com/Automattic/wp-calypso/blob/1e86a84f3c3f715e6b7c6917acb695f8206b8c41/client/gutenboarding/onboarding-block/index.ts#L14-L18.

Probably fine to use `string` for a first iteration, although there's a bit of a risk we might forget to change it later, and it's arguably one case where we see TypeScript forcing/helping us to be consistent with types.